src/components/BoxForm/BoxForm.jsx

```diff
@@ -0,0 +1,234 @@
+import FYABackend from '../../common/utils';
+import DropZone from './DropZone/DropZone';
+import './BoxForm.css';
+
+function Box() {
+  const [submit, setSubmit] = useState(false);
+  const [files, setFiles] = useState([]);
+  const [formData, setFormData] = useState({
+    boxNumber: '',
+    date: '',
+    zipCode: '',
+    boxLocation: '',
+    message: '',
+    picture: '',
+    comments: '',
+  });
+
+  // states for tracking error inputs
+  const [zipCodeError, setZipCodeError] = useState(false);
+  const [dateError, setDateError] = useState(false);
+  const [boxNumberError, setBoxNumberError] = useState(false);
+
+  // useEffect ensures that the most recently updated formData is reflected
+  useEffect(async () => {
+    if (submit) {
+      // send formdata to server
+      const res = await FYABackend.post('/boxForm', formData, {
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      });
+      console.log(res.data);
+      setSubmit(false);
+    }
+  }, [submit]);
+
+  const { boxNumber, date, zipCode, boxLocation, message, comments } = formData;
+
+  const uploadBoxPhoto = async file => {
+    // get S3 upload url from server
+    const { data: uploadUrl } = await FYABackend.get('/s3Upload');
+
+    // upload image directly to S3 bucket
+    await axios.put(uploadUrl, file, {
+      headers: {
+        'Content-Type': 'multipart/form-data',
+      },
+    });
+
+    // return box image url
+    const imageUrl = uploadUrl.split('?')[0];
+    return imageUrl;
+  };
+
+  const isValidZip = zip => {
+    const countries = [
+      'US',
+      'AT',
+      'BG',
+      'BR',
+      'CA',
+      'CZ',
+      'DK',
+      'FR',
+      'DE',
+      'IN',
+      'IT',
+      'IE',
+      'MA',
+      'NL',
+      'PL',
+      'PT',
+      'RO',
+      'RU',
+      'SG',
+      'SK',
+      'ES',
+      'SE',
+      'CH',
+      'GB',
+    ];
+    return countries.filter(country => isValidZipcode(zip, country)).length > 0;
+  };
+
+  const onChange = e => {
+    setFormData({ ...formData, [e.target.name]: e.target.value });
+  };
+
+  const submitForm = async () => {
+    if (files.length > 0) {
+      const imageUrl = await uploadBoxPhoto(files[0]);
+      setFormData({ ...formData, picture: imageUrl });
+    }
+    setSubmit(true);
+  };
+
+  const onSubmit = e => {
+    e.preventDefault();
+    setBoxNumberError(boxNumber === '' && true);
+    setZipCodeError((zipCode === '' || !isValidZip(zipCode)) && true);
+    setDateError(date === '' && true);
+    if (!(boxNumberError || dateError || zipCodeError)) {
+      submitForm();
+    }
+  };
+
+  return (
+    <form className="boxForm" onSubmit={e => onSubmit(e)}>
+      <div className="a">
+        <div className="boxInfo">
+          <FormControl isInvalid={dateError}>
+            <FormLabel htmlFor="date">Date</FormLabel>
+            <Input type="date" id="date" name="date" value={date} onChange={e => onChange(e)} />
```

Comment:
Can we use React date picker here instead of the input tag? I believe the input date may not work on all browsers.
https://yarnpkg.com/package/react-datepicker